src/storm/modelchecker/helper/conditional/ConditionalHelper.cpp

```diff
@@ -0,0 +1,772 @@
+        return;
+    }
+    auto const reachableStates = storm::utility::graph::getReachableStates(transitionMatrix, initialStates, allowedStates, targetStates);
+    auto const subTargets = targetStates % reachableStates;
+    // Catch the case where no target is reachable from an initial state. In this case, there is nothing to do since all probabilities are zero.
+    if (subTargets.empty()) {
+        return;
+    }
+    auto const subInits = initialStates % reachableStates;
+    auto const submatrix = transitionMatrix.getSubmatrix(true, reachableStates, reachableStates);
+    auto const subResult = helper::SparseMdpPrctlHelper<ValueType, ValueType>::computeUntilProbabilities(
+        env, storm::solver::SolveGoal<ValueType>(dir, subInits), submatrix, submatrix.transpose(true), storm::storage::BitVector(subTargets.size(), true),
+        subTargets, false, false);
+    auto origInitIt = initialStates.begin();
+    for (auto subInit : subInits) {
+        auto const& val = subResult.values[subInit];
+        if (!storm::utility::isZero(val)) {
+            nonZeroResults.emplace(*origInitIt, val);
+        }
+        ++origInitIt;
+    }
+}
+
+template<typename ValueType>
+struct NormalFormData {
+    storm::storage::BitVector const maybeStates;      // Those states that can be reached from initial without reaching a terminal state
+    storm::storage::BitVector const terminalStates;   // Those states where we already know the probability to reach the condition and the target value
+    storm::storage::BitVector const conditionStates;  // Those states where the condition holds almost surely (under all schedulers)
+    storm::storage::BitVector const universalObservationFailureStates;    // Those states where the condition is not reachable (under all schedulers)
+    storm::storage::BitVector const existentialObservationFailureStates;  // Those states s where a scheduler exists that (i) does not reach the condition from
+                                                                          // s and (ii) acts optimal in all terminal states
+    std::map<uint64_t, ValueType> const nonZeroTargetStateValues;         // The known non-zero target values. (default is zero)
+    // There are three cases of terminal states:
+    // 1. conditionStates: The condition holds, so the target value is the optimal probability to reach target from there
+    // 2. targetStates: The target is reached, so the target value is the optimal probability to reach a condition from there.
+    //                  The remaining probability mass is the probability of an observation failure
+    // 3. states that can not reach the condition under any scheduler. The target value is zero.
+
+    // TerminalStates is a superset of conditionStates and dom(nonZeroTargetStateValues).
+    // For a terminalState that is not a conditionState, it is impossible to (reach the condition and not reach the target).
+
+    ValueType targetValue(uint64_t state) const {
```

Comment:
I know this is more c++ style than what we usually do, but would expect in this code base that this is called getTargetValue?